ride-share.rb

```diff
@@ -0,0 +1,107 @@
+##### RIDE SHARE PROGRAM #####
+
+# place to store all ride data
+rides = {
+    DR0001: 
+    [
+        {date: "3rd Feb 2016", cost: 10, rider_id: "RD0003", rating: 3},
+        {date: "3rd Feb 2016", cost: 30, rider_id: "RD0015", rating: 4},
+        {date: "5th Feb 2016", cost: 45, rider_id: "RD0003", rating: 2}
+    ],
+
+    DR0002:
+    [
+        {date: "3rd Feb 2016", cost: 25, rider_id: "RD0073", rating: 5},
+        {date: "4th Feb 2016", cost: 15, rider_id: "RD0013", rating: 1},
+        {date: "5th Feb 2016", cost: 35, rider_id: "RD0066", rating: 3}
+    ],
+
+    DR0003:
+    [
+        {date: "4th Feb 2016", cost: 5, rider_id: "RD0066", rating: 5},
+        {date: "5th Feb 2016", cost: 50, rider_id: "RD0003", rating: 2}
+    ],
+
+    DR0004:
+    [
+        {date: "3rd Feb 2016", cost: 5, rider_id: "RD0022", rating: 5},
+        {date: "4th Feb 2016", cost: 10, rider_id: "RD0022", rating: 4},
+        {date: "5th Feb 2016", cost: 20, rider_id: "RD0073", rating: 5}
+    ]
+}
+
+# counts total number of rides complete per driver
+def total_rides_by_driver(rides)
+    total_rides = {}
+
+    rides.each do |driver, ride_data|
+        total_rides[driver] = ride_data.length
+    end
+    return total_rides
+end
+
+# sums total amount of money made per driver
+def total_money_by_driver(rides)
+    total_money = {}
+    
+    rides.each do |driver, ride_data| 
+        ride_cost = ride_data.map { |hash| hash[:cost] }
+        total_money[driver] = ride_cost.sum
+    end
+    return total_money
+end
+
+# averages ratings per driver
+def average_rating_by_driver(rides)
+    average_rating = {}
+
+    rides.each do |driver, ride_data| 
+        ride_rating = ride_data.map { |hash| hash[:rating] }
```

Comment:
In this case, `hash` is the name of an element, which represents something in `ride_data`. It might be better to have a different name besides `hash`!